### PR TITLE
CSS Fixes for md-dialog

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Looks like during the conversion to the new dialog some invalid CSS sneaked in.

I noticed it because I was building the demo locally, and the HTML minifier complained about invalid CSS.

I've not tested this PR. Please test and verify that it still works as intended (or feel free to close it / remove the CSS that probably didn't work before)

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by wendevlin — CHANGES_REQUESTED** [review]

File: src/components/ha-md-dialog.ts [hunk 1]
```diff
@@ -178,15 +178,15 @@ export class HaMdDialog extends MdDialog {
         }
       }
 
-      :host ::slotted(ha-dialog-header) {
+      ::slotted(ha-dialog-header) {
```
[the remark this answers is not in the document]
Comment (balloob) [reply src/components/ha-md-dialog.ts:181]: Everything is inside the host though? What does this selector do?

Take a look at this playground: https://developer.mozilla.org/en-US/docs/Web/CSS/:host#try_it

Change the CSS from `:host` to `:host h1` and the selector no longer works

[the remark this answers is not in the document]
Comment (balloob) [reply src/components/ha-md-dialog.ts:181]: You can use `:host` or you can match a host with a specific class/attribute, like `:host(.warning)`

[the remark this answers is not in the document]
Comment (balloob) [reply src/components/ha-md-dialog.ts:181]: Docs
- https://developer.mozilla.org/en-US/docs/Web/CSS/:host
- https://developer.mozilla.org/en-US/docs/Web/CSS/:host-context
- https://developer.mozilla.org/en-US/docs/Web/CSS/:host_function

[the remark this answers is not in the document]
Comment (wendevlin) [reply src/components/ha-md-dialog.ts:181]: 😄 🤷 I can just say when you remove it the ha-dialog-header is broken inside an md-dialog. 

I will have a detailed look into it tomorrow.

[the remark this answers is not in the document]
Comment (wendevlin) [reply src/components/ha-md-dialog.ts:181]: I edited the suggestion, when we specify the slot it work. We use `:host ::slotted` multiple times in the frontend project, it seems to be a special case for `::slotted` and how it overrides global rules.

[the remark this answers is not in the document]
Comment (wendevlin) [reply src/components/ha-md-dialog.ts:181]: @balloob is this a fine solution for you?

[the remark this answers is not in the document]
Comment (balloob) [reply src/components/ha-md-dialog.ts:181]: Sure, if you say it works then let's do it 👍

**Review by wendevlin — APPROVED** [review]

**Pull request merged**